packages/shared/src/components/fields/CodeField.tsx

```diff
@@ -0,0 +1,66 @@
+import type { ReactElement, ClipboardEventHandler, ChangeEvent } from 'react';
+import React, { useRef, useState } from 'react';
+import { TextField } from './TextField';
+
+interface CodeFieldProps {
+  onChange: (code: string) => void;
+  length?: number;
+}
+
+const DEFAULT_LENGTH = 6;
+
+export function CodeField({
+  onChange,
+  length = DEFAULT_LENGTH,
+}: CodeFieldProps): ReactElement {
+  const elementsRef = useRef<HTMLInputElement[]>(Array(length).fill(null));
+  const [code, setCode] = useState<string[]>(Array(length).fill(''));
+
+  const onPaste: ClipboardEventHandler<HTMLInputElement> = (e) => {
+    const text = e.clipboardData.getData('text');
+    const sliced = text.slice(0, DEFAULT_LENGTH);
+    setCode(sliced.split(''));
+    onChange(sliced);
+  };
+
+  const onInputChange = (e: ChangeEvent<HTMLInputElement>, index) => {
+    const char = e.target.value;
+    const newCode = [...code];
+    newCode[index] = char;
+    setCode(newCode);
+
+    if (char.length > 0 && index < DEFAULT_LENGTH - 1) {
+      const nextIndex = index + 1;
+
+      if (elementsRef.current[nextIndex]) {
+        elementsRef.current[nextIndex].focus();
+      }
+    }
```

Comment:
Testing it now, it does not work on backspace, so if you press it it should remove and focus on previous box

Reply:
Yeah, because maybe you only need to update 1 box hence we dont wanna go back.

Reply:
What do you mean?

Reply:
Like if the code you typed was wrong, but there was only 1 wrong number. When you click the wrong box and then click delete, if we move the focus, you will go back to the other field instead of the one you just deleted and needed to update.

Reply:
Usually what happens is that when you click on input and press backspace (or any other character) it first replaces what you have and then goes back. Something like this https://codesandbox.io/p/sandbox/2fa-input-in-react-ltgjh?file=%2Fsrc%2FApp.js

Not blocking but just think backspace is just a standard thing that works in this kind of inputs, especially for people only using keyboards.

Reply:
Yeah, we can definitely have that. I just vaguely remembered one time I was typing OTP on an app and focus wasnt changing after backspace. But because maybe it was on mobile.